docs/spectrum/simulation.rst

```diff
@@ -0,0 +1,103 @@
+.. include:: ../references.txt
+
+.. _spectrum_simulation:
+
+******************
+Simulating Spectra 
+******************
+
+.. currentmodule:: gammapy.spectrum
+
+In the following, a few example are given to demonstrate the
+`~gammapy.spectrum.SpectrumSimulation` class.
+
+Getting Started
+===============
+
+This section show you how to simulate a `~gammapy.spectrum.SpetrumObservation`
+given a source model and instrument response functions. The latter will also be
+simulated to make this tutorial independent of any real data. No background
+measurement is simulated in this example. Simulating the IRFs works like this
+
+.. code-block:: python
+
+    import numpy as np
+    import astropy.units as u
+    from gammapy.irf import EnergyDispersion, EffectiveAreaTable
+        
+    e_true = np.logspace(-2, 2.5, 109) * u.TeV
+    e_reco = np.logspace(-2,2, 79) * u.TeV
+
+    edisp = EnergyDispersion.from_gauss(e_true=e_true, e_reco=e_reco, sigma=0.2)
+    aeff = EffectiveAreaTable.from_parametrization(energy=e_true)
+
+We'll have a look at the simulated IRFs in a second. But first we define a
+`~gammapy.spectrum.models.SpectralModel`,
+
+
+.. code-block:: python
+
+    from gammapy.spectrum import models
+
+    index = 2.3 * u.Unit('')
+    amplitude = 2.5 * 1e-12 * u.Unit('cm-2 s-1 TeV-1')
+    reference = 1 * u.TeV
+
+    model = models.PowerLaw(index=index, amplitude=amplitude, reference=reference) 
+    
+and simulate one `~gammapy.spectrum.SpectrumObservation` for one 4 hour
+observation. We set the low energy threshold to 20% of the peak effective area
+and the high energy threshold to 60 TeV.
+
+.. code-block:: python
+
+    from gammapy.spectrum import SpectrumSimulation
+    import matplotlib.pyplot as plt
+
+    livetime = 4 * u.h
+    lo_threshold = aeff.find_energy(0.2 * aeff.max_area)
+    hi_threshold = 60 * u.TeV
+
+    sim = SpectrumSimulation(aeff=aeff, edisp=edisp, model=model, livetime=livetime)
+    
+    obs = sim.simulate_obs(obs_id=42, seed=42,
+                           lo_threshold=lo_threshold,
+                           hi_threshold=hi_threshold)
+    obs.peek()
+    plt.show()
+
+.. image:: simulated_obs.png
+
+
+Simulating Many Spectra
+=======================
+
+To get a more quantitative impression of the
+`~gammapy.spectrum.SpectrumSimulation` we are now going to simulate ``n``
+spectra, fit a model to them and compare the model parameter estimates with the
+input values. Please read :ref:`spectral_fitting` if you want to know how to
+fit a `~gammapy.spectrum.SpectrumObservation`. We'll use the model and the
+simulated IRFs from the Getting Started section in the following.
+
+
+.. code-block:: python
+
+    # mute the sherpa logger
+    import sherpa
+    import logging
+
+    sherpa_logger = logging.getLogger('sherpa')
+    sherpa_logger.setLevel(50)
+
+    n = 100
+    index_estimate = list()
+    amplitude_estimate = list()
```

Comment:
I usually use list of dict instead of parallel lists to collect data.
I find it easier to read / extend / maintain.
Example: https://github.com/gammapy/gammapy/blob/b6d3671a2ecdc36f6c12dcee6f5098c137d1c123/gammapy/datasets/core.py#L182